server/src/main/java/org/elasticsearch/plugins/NetworkPlugin.java

```diff
@@ -75,7 +75,8 @@
                                                                          NamedXContentRegistry xContentRegistry,
                                                                          NetworkService networkService,
                                                                          HttpServerTransport.Dispatcher dispatcher,
-                                                                         ClusterSettings clusterSettings) {
+                                                                         ClusterSettings clusterSettings,
+                                                                         RestCompatibility restCompatibleFunction) {
```

Comment:
nit: `Function` in the name maybe a bit redundant (here and elsewhere).  If you change the functional interface name, it may allow this to read better.  `CompatibleVersion compatibleVersion` , and a simpler 	`compibleVersion.get(a,b,c)` ?